### ISSUE TITLE
Babel queries for loaders

### ISSUE THREAD

**Issue by mmahalwy** [amireh/happypack#65]
I cannot get happypack to work when I have:
```
{ 
        test: /\.js$/,
        exclude: /node_modules/,
        loaders: [
          {
            loader: 'babel',
            query: {
              plugins: [
                'transform-runtime',
                'add-module-exports',
                'transform-decorators-legacy',
                'transform-react-display-name',
                'typecheck',
                'transform-react-jsx-self',
                'transform-react-jsx-source'
              ],
              presets: ['react', 'es2015-webpack', 'stage-0', 'react-hmre'],
              cacheDirectory: true
            }
          },
          'eslint-loader'
        ],
      },
```

Errors with:
```
/Users/mmahalwy/Sites/sparrow/node_modules/happypack/lib/WebpackUtils.js:9
  var loaderStrings = string.split('!');
                             ^

TypeError: string.split is not a function
    at disectLoaderString (/Users/mmahalwy/Sites/sparrow/node_modules/happypack/lib/WebpackUtils.js:9:30)
    at Array.map (native)
    at normalizeLoaders (/Users/mmahalwy/Sites/sparrow/node_modules/happypack/lib/HappyPlugin.js:171:10)
    at /Users/mmahalwy/Sites/sparrow/node_modules/happypack/node_modules/async/lib/async.js:713:13
    at Immediate.iterate (/Users/mmahalwy/Sites/sparrow/node_modules/happypack/node_modules/async/lib/async.js:262:13)
    at runCallback (timers.js:570:20)
    at tryOnImmediate (timers.js:550:5)
    at processImmediate [as _immediateCallback] (timers.js:529:5)
```

If I move it to the `new HappyPack({ id: 'js', loaders: ...})` if throws error:
```
assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: HappyPack[js]: Loader must have a @path property or be a string.
    at /Users/mmahalwy/Sites/sparrow/node_modules/happypack/lib/OptionParser.js:31:7
    at Array.reduce (native)
    at parseAndValidateOptions (/Users/mmahalwy/Sites/sparrow/node_modules/happypack/lib/OptionParser.js:19:30)
    at new HappyPlugin (/Users/mmahalwy/Sites/sparrow/node_modules/happypack/lib/HappyPlugin.js:33:17)
    at Object.<anonymous> (/Users/mmahalwy/Sites/sparrow/webpack/dev.config.js:69:5)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
```

**Comment by amireh** [comment]
@mmahalwy try running against the master branch and see if the issue is solved. Once you confirm we can make another release. Thanks!

**Comment by mmahalwy** [comment]
works great! Thank you!

**Comment by amireh** [comment]
The changes are published in the new release 2.1.3, closing this now!
